docs/tool/js/views/map-view.js

```diff
@@ -869,6 +871,59 @@
                 .remove();
 
         },
+        exportCsv: function() {
+          // if (data){
+            d3.select('#csvExport')
+              .on('click', function(d) {
+                  console.log("YOU CLICKED EXPORT CSV");
+                  var preview = d3.select('#buildings-list');
+                  var listItems = preview.selectAll('div');
+                  var projectNames = [];
+                  listItems._groups[0].forEach( function(project){
+                    projectNames.push(project.innerText);
+                  })
+                  var allData = mapView.convertedProjects.features;
+                  // console.log(mapView.convertedProjects);
+                  var matchesData = allData.filter(function(feature) {
+                      return feature.properties.matches_filters === true;
+                  });
+                  var notMatchesData = allData.filter(function(feature) {
+                      return feature.properties.matches_filters === false;
+                  });
+
+                  //Create a csv from the data
+                  var csvContent = "data:text/csv;charset=utf-8,";
+                  var keys = "matches_filters,";
+                  for (key in allData[0].properties){
+                    if ( key !== "matches_filters"){
+                      keys += key + ",";
+                    }
+                  }
+                  csvContent += keys + '\n';
+                  matchesData.forEach( function(project){
+                    csvContent += String(project.properties['matches_filters']) + ",";
+                    for (key in project.properties){
+                      if ( key !== "matches_filters"){
+                        csvContent += String(project.properties[key]).replace(/,/g,' ') + ",";
+                      }
+                    }
+                    csvContent += '\n';
+                  })
+                  notMatchesData.forEach( function(project){
+                    csvContent += String(project.properties['matches_filters']) + ",";
+                    for (key in project.properties){
+                      if ( key !== "matches_filters"){
+                        csvContent += String(project.properties[key]).replace(/,/g,' ') + ",";
+                      }
+                    }
+                    csvContent += '\n';
+                  })
+                  var encodedUri = encodeURI(csvContent);
+                  window.open(encodedUri);
```

Comment:
Can you have it provide a default name for the file to be downloaded? At a minimum, we need it to specify the .csv file type automatically so that users won't need to add the ending themselves - many users won't know what to do with the file if it doesn't come with the right extension on it.

Reply:
Done. projects.csv